src/box/tuple_hash.cc

```diff
@@ -285,9 +371,10 @@ key_def_set_hash_func(struct key_def *key_def) {
 	key_def->key_hash = key_hash_slowpath;
 }
 
-uint32_t
-tuple_hash_field(uint32_t *ph1, uint32_t *pcarry, const char **field,
-		 enum field_type type, struct coll *coll)
+template<tuple_hash_version version>
```

Comment:
Do we really want to make this function support two formats? Since it's used only by the bloom filter, wouldn't it be better to copy the legacy implementation to `tuple_bloom.c`?

Reply:
Nope, I didn't want it to support 2 formats. But I couldn't find a good solution here. The motivation is that either it is like this, or we are going to have some hashing functions duplicated with no test coverage, or we would need to introduce test coverage for them.

I decided the template would be the least intrusive solution while still keeping those legacy hash functions at least partially covered by tests.

Do you think it is better to duplicate? Do we need to also duplicate the tests then?

Reply:
I would prefer to just copy the old implementation to `tuple_bloom.c`. Code duplication is fine in this case IMO because we'll eventually drop them. I don't think we need to test these functions specifically because we aren't ever going to extend/patch them so the tests checking the legacy bloom filter functionality in Vinyl should be enough. Let's just add a few more indexes with some data to `gh_9965_data/gen.lua`  (double, number, string) to formally cover the code computing the legacy hash. For example, create an index storing number from 1 to 10 with step 0.5 and check that `get` works as expected.